test/dummy/test/helpers/store_accessor_test_helper.rb

```diff
@@ -90,4 +90,127 @@ def test_define_all_store_accessors_works_correctly
     assert_respond_to instance, :party_theme
     assert_respond_to instance, :party_theme=
   end
+
+  def test_setting_store_attributes_with_new
+    # Test setting store attributes via new with mass assignment
+    record = klass.new(
+      store_versioned_schema: structured_store_versioned_schemas(:party),
+      party_theme: 'Hawaiian',
+      balloon_count: 25
+    )
+
+    assert_equal 'Hawaiian', record.party_theme
+    assert_equal 25, record.balloon_count
+    assert_not record.persisted?
+
+    # Save and verify persistence
+    record.save!
+    assert record.persisted?
+
+    # Reload and verify values were saved correctly
+    reloaded_record = klass.find(record.id)
+    assert_equal 'Hawaiian', reloaded_record.party_theme
+    assert_equal 25, reloaded_record.balloon_count
+  end
+
+  def test_setting_store_attributes_with_create
+    # Test setting store attributes via create with mass assignment
+    record = klass.create(
+      store_versioned_schema: structured_store_versioned_schemas(:party),
+      party_theme: 'Masquerade',
+      balloon_count: 50
+    )
+
+    assert record.persisted?
+    assert_equal 'Masquerade', record.party_theme
+    assert_equal 50, record.balloon_count
+
+    # Reload and verify values persisted
+    reloaded_record = klass.find(record.id)
+    assert_equal 'Masquerade', reloaded_record.party_theme
+    assert_equal 50, reloaded_record.balloon_count
+  end
+
+  def test_setting_store_attributes_with_create_bang
+    # Test setting store attributes via create! with mass assignment
+    record = klass.create!(
+      store_versioned_schema: structured_store_versioned_schemas(:party),
+      party_theme: 'Carnival',
+      balloon_count: 100
+    )
+
+    assert record.persisted?
+    assert_equal 'Carnival', record.party_theme
+    assert_equal 100, record.balloon_count
+
+    # Reload and verify values persisted
+    reloaded_record = klass.find(record.id)
+    assert_equal 'Carnival', reloaded_record.party_theme
+    assert_equal 100, reloaded_record.balloon_count
+  end
+
+  def test_setting_store_attributes_without_schema_with_new
```

Comment:
Is it worth an additional test that still raises `ActiveModel::UnknownAttributeError`, when a schema is provided but it doesn't define a provided attribute?


```ruby 
  def test_setting_store_attributes_with_undefined_attribute
    assert_raises(ActiveModel::UnknownAttributeError) do
      klass.new(
        store_versioned_schema: structured_store_versioned_schemas(:party),
        party_theme: 'Pirates',
        party_location: 'White Hart Lane',
        balloon_count: 10
      )
    end
  end
```